modules/ztp-image-based-upgrade-concept.adoc

```diff
@@ -0,0 +1,147 @@
+The first action of an added step for clusters that failed a previous steps must be either `Abort` or `Rollback`.
+
+[IMPORTANT]
+====
+You cannot remove actions or steps from an ongoing plan.
+====
+
+The following table shows example plans for different levels of control over the rollout strategy:
+
+.Example upgrade plans
+[cols=2*, width="100%", options="header"]
+|====
+|Example plan
+|Description
+
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep", "Upgrade", "FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+----
+|All actions share the same strategy
+
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep", "Upgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+- actions: ["FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 500
+    timeout: 10
+----
+|Some actions share the same strategy
+
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+- actions: ["Upgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 20
+- actions: ["FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 500
+    timeout: 10
+----
+|All actions have different strategies
+
+|====
+
+[IMPORTANT]
+====
+Clusters that fail one of the actions will skip the remaining actions in the same step.
+====
+
+The `ImageBasedGroupUpgrade` API accepts the following actions:
+
+`Prep`:: Start preparing the upgrade resources by moving to the `Prep` stage.
+`Upgrade`:: Start the upgrade by moving to the `Upgrade` stage.
+`FinalizeUpgrade`:: Finalize the upgrade on selected clusters that completed the `Upgrade` action by moving to the `Idle` stage.
+`Rollback`:: Start a rollback only on successfully upgraded clusters by moving to the `Rollback` stage.
+`FinalizeRollback`:: Finalize the rollback by moving to the `Idle` stage.
+`AbortOnFailure`:: Cancel the upgrade on selected clusters that failed the `Prep` or `Upgrade` actions by moving to the `Idle` stage.
+`Abort`:: Cancel an ongoing upgrade only on clusters that are not yet upgraded by moving to the `Idle` stage.
+
+The following action combinations are supported. A pair of brackets signifies one step in the `plan` section:
+
+* `["Prep"]`, `["Abort"]`
+* `["Prep", "Upgrade", "FinalizeUpgrade"]`
+* `["Prep"]`, `["AbortOnFailure"]`, `["Upgrade"]`, `["AbortOnFailure"]`, `["FinalizeUpgrade"]`
+* `["Rollback", "FinalizeRollback"]`
+
+Use one of the following combinations when you need to resume or cancel an ongoing upgrade from a completely new `ImageBasedGroupUpgrade` CR:
+
+* `["Upgrade","FinalizeUpgrade"]`
+* `["FinalizeUpgrade"]`
+* `["FinalizeRollback"]`
+* `["Abort"]`
+* `["AbortOnFailure"]`
+
+[id="ztp-image-based-upgrade-cluster-labeling_{context}"]
+== Labeling for cluster selection
+
+Use the `spec.clusterLabelSelectors` field for initial cluster selection.
+In addition, {cgu-operator} labels the managed clusters according to the results of their last stage transition.
```

Comment:
```suggestion
In addition, the {cgu-operator} labels the managed clusters according to the results of their last stage transition.
```

For consistency, as per https://github.com/openshift/openshift-docs/pull/86756/files#diff-31c7701b2c54c828ab2e3cbb1ca1b42dd00f088e36e57ad8a57cbe3c39978181R130